api/v1beta1/common_types.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2023 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// HeatTemplate -
+type HeatTemplate struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=heat
+	// ServiceUser - optional username used for this service to register in heat
+	ServiceUser string `json:"serviceUser"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=heat
+	// DatabaseUser - optional username used for heat DB, defaults to heat.
+	// TODO: -> implement needs work in mariadb-operator, right now only heat.
+	DatabaseUser string `json:"databaseUser"`
+
+	// +kubebuilder:validation:Optional
+	// DatabaseHostname - Heat Database Hostname
+	DatabaseHostname string `json:"databaseHostname,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// Secret containing OpenStack password information for heat HeatDatabasePassword, HeatPassword
+	// and HeatAuthEncryptionKey
+	Secret string `json:"secret"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default={database: HeatDatabasePassword, service: HeatPassword, authEncryptionKey: HeatAuthEncryptionKey}
+	// PasswordSelectors - Selectors to identify the DB and ServiceUser password from the Secret
+	PasswordSelectors PasswordSelector `json:"passwordSelectors,omitempty"`
+}
+
+// HeatServiceTemplate -
+type HeatServiceTemplate struct {
+	// +kubebuilder:validation:Required
+	// ContainerImage - Heat API Container Image URL
+	ContainerImage string `json:"containerImage"`
+
+	// +kubebuilder:validation:Optional
+	// Debug - enable debug for different deploy stages. If an init container is used, it runs and the
+	// actual action pod gets started with sleep infinity
+	Debug HeatServiceDebug `json:"debug,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Maximum=32
+	// +kubebuilder:validation:Minimum=0
+	// Replicas -
+	Replicas int32 `json:"replicas"`
+
+	// +kubebuilder:validation:Optional
+	// NodeSelector to target subset of worker nodes for running the API service
```

Comment:
We also use this `HeatServiceTemplate` for HeatEngine, so it's not just API services. But it's a small nit